### ISSUE TITLE
Allow wildcard field parameters

### ISSUE THREAD

**Issue by amitailanciano** [Propaganistas/Laravel-Phone#34]
Thanks again for putting this out there.

Laravel 5.2 now supports validation arrays, it looks like the phone validator doesn't take advantage of that yet, I'd like to be able to do this:

```
'something.*.phone' => 'phone:something.*.phone_country'
```

Similar to what is shown in the docs:

```
$validator = Validator::make($request->all(), [
    'person.*.email' => 'email|unique:users',
    'person.*.first_name' => 'required_with:person.*.last_name',
]);
```

https://laravel.com/docs/5.2/validation#validating-arrays

**Comment by Propaganistas** [comment]
I'd love to, but at the moment Laravel doesn't allow third-party packages to implement wildcards in validator parameters...

Parsing of wildcard parameters happens here:
https://github.com/laravel/framework/blob/5.2/src/Illuminate/Validation/Validator.php#L473
In order for this line to fire, the `dependsOnOtherFields()` conditional should return true. This method however checks the `$dependentFields` property which, unfortunately, cannot be modified from the outside...

**Comment by Propaganistas** [comment]
This has been implemented as of Laravel 5.4.18

https://github.com/laravel/framework/pull/18654

**Comment by Propaganistas** [comment]
Implemented. Please update to the latest version (and make sure to use Laravel >= 5.4.18):

https://github.com/Propaganistas/Laravel-Phone/releases/tag/2.8.3
